api/src/main/java/org/apache/iceberg/actions/AnalyzeTable.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.util.List;
+import java.util.Set;
+
+/** An action that collects statistics of an Iceberg table and writes to Puffin files. */
+public interface AnalyzeTable extends Action<AnalyzeTable, AnalyzeTable.Result> {
+  /**
+   * The set of columns to be analyzed
+   *
+   * @param columns a set of column names to be analyzed
+   * @return this for method chaining
+   */
+  AnalyzeTable columns(Set<String> columns);
+
+  /**
+   * A set of statistics to be collected on the given columns of the given table
+   *
+   * @param statsToBeCollected set of statistics to be collected
+   * @return this for method chaining
+   */
+  AnalyzeTable stats(Set<String> statsToBeCollected);
```

Comment:
Should these stats be a `Set<StandardBlobType>` instead of arbitrary Strings? I feel like the API becomes more well defined in this case.

Reply:
Oh I see, StandardBlobType defines string constants not enums